common/test/acceptance/tests/studio/test_studio_library.py

```diff
@@ -343,6 +343,65 @@ def test_delete_shifts_blocks(self):
         self.assertEqual(self.lib_page.xblocks[-1].name, '11')
         self.assertEqual(self.lib_page.get_page_number(), '1')
 
+    def test_previews(self):
+        """
+        Scenario: Ensure the user is able to hide previews of XBlocks.
+        Given that I have a library in Studio with 40 XBlocks
+        Then previews are visible
+        When I click the toggle previews button
+        The previews will not be visible
+        When I click the toggle previews button
+        The previews are visible
+        """
+        self.assertTrue(self.lib_page.are_previews_showing())
+        self.lib_page.toggle_previews()
+        self.assertFalse(self.lib_page.are_previews_showing())
+        self.lib_page.toggle_previews()
+        self.assertTrue(self.lib_page.are_previews_showing())
+
+    def test_previews_navigation(self):
+        """
+        Scenario: Ensure preview settings persist across navigation.
+        Given that I have a library in Studio with 40 XBlocks
+        Then previews are visible
+        When I click the toggle previews button
+        And click the next page button
+        The previews will not be visible
+        And the first XBlock will be the 11th one
+        And the last XBlock will be the 20th one
+        When I click the toggle previews button
+        And I click the previous page button
+        The previews will be visible
+        And the first XBlock will be the first one
+        And the last XBlock will be the 11th one
+        """
+        self.assertTrue(self.lib_page.are_previews_showing())
+        self.lib_page.toggle_previews()
+        # Which set of arrows shouldn't matter for this test.
+        self.lib_page.move_forward('top')
+        self.assertFalse(self.lib_page.are_previews_showing())
+        self.assertEqual(self.lib_page.xblocks[0].name, '11')
+        self.assertEqual(self.lib_page.xblocks[-1].name, '20')
+        self.lib_page.toggle_previews()
+        self.lib_page.move_back('top')
+        self.assertTrue(self.lib_page.are_previews_showing())
+        self.assertEqual(self.lib_page.xblocks[0].name, '1')
+        self.assertEqual(self.lib_page.xblocks[-1].name, '10')
+
+    def test_preview_state_persistance(self):
+        """
+        Scenario: Ensure preview state persists between page loads.
+        Given that I have a library in Studio with 40 XBlocks
+        Then previews are visible
+        When I click the toggle previews button
+        And I revisit the page
+        The previews will not be visible
+        """
+        self.assertTrue(self.lib_page.are_previews_showing())
+        self.lib_page.toggle_previews()
+        self.lib_page.visit()
+        self.assertFalse(self.lib_page.are_previews_showing())
```

Comment:
Add tests for expected behavior when previews are hidden and:

1) A new component is added
2) An existing component is edited